### PR TITLE
Migrate to all newest packages, except for eslint v9

### PR DESCRIPTION
**Pull request by zglicz**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-sonarjs",
-  "version": "0.25.1",
+  "version": "0.26.0",
```
Comment (zglicz) [inline package.json:3]: I didn't land this version pull after the last release (https://github.com/SonarSource/eslint-plugin-sonarjs/pull/455)

File: tsconfig.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     "allowSyntheticDefaultImports": true,
     "noEmit": false,
     "outDir": "lib",
-    "sourceMap": true
+    "sourceMap": true,
+    "moduleResolution": "nodenext"
```
Comment (zglicz) [inline tsconfig.json:14]: This is required to allow for the typescript-eslint libs to work. See more details - https://github.com/typescript-eslint/typescript-eslint/issues/7284

File: tests/rule-tester.ts [hunk 3]
```diff
@@ -18,9 +18,7 @@
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
 
-import { ESLintUtils } from '@typescript-eslint/utils';
-
-const { RuleTester } = ESLintUtils;
+import { RuleTester } from '@typescript-eslint/rule-tester';
```
Comment (zglicz) [inline tests/rule-tester.ts:21]: Breaking change, now lives in a separate package - https://typescript-eslint.io/packages/rule-tester/

File: package-lock.json
Comment (ericmorand-sonarsource) [inline package-lock.json:1]: The URL to the jFrog repository should be replaced by the official npm repository - which means "use the repository that is configured by default for the npm client".

I usually replace every occurrence of `https://repox.jfrog.io/artifactory/api/npm/npm` with `https://registry.npmjs.org`.

File: tests/index.test.ts [hunk 4]
```diff
@@ -29,6 +29,7 @@ it('should declare all rules in recommended config', () => {
     expect(configs.recommended.rules).toHaveProperty(`sonarjs/${rule}`);
   });
   expect(Object.keys(configs.recommended.rules!)).toHaveLength(existingRules.length);
+  expect(new Set(Object.values(configs.recommended.rules!))).toEqual(new Set(['off', 'error']));
```
Comment (zglicz) [inline tests/index.test.ts:32]: I added this check, as I would have missed almost exporting "recommended"

File: src/rules/cognitive-complexity.ts [hunk 5]
```diff
@@ -61,13 +62,13 @@ const rule: TSESLint.RuleModule<string, (number | 'metric' | 'sonar-runtime')[]>
     type: 'suggestion',
     docs: {
       description: 'Cognitive Complexity of functions should not be too high',
-      recommended: 'error',
```
Comment (zglicz) [inline src/rules/cognitive-complexity.ts:64]: omitting this value, implies it's not recommended to be turned on. And also, based on this value, we compile our plugin.

**Review by zglicz — COMMENTED** [review]
PR should be ready for review. SonarCloud analysis is passing, no more deprecated API's, tests passing and the build issue is unrelated and tracked in: https://sonarsource.atlassian.net/issues/BUILD-4777?jql=ORDER%20BY%20created%20DESC

**Comment by ericmorand-sonarsource** [comment]
Looks good to me: tests pass locally and build succeeds. As soon as the CI is fixed we can merge IMHO.

**Review by ericmorand-sonarsource — APPROVED** [review]

**Pull request merged**